source/unified-test-format/unified-test-format.rst

```diff
@@ -1798,6 +1843,134 @@ Use a ``listIndexes`` command to check whether the index exists. Note that it is
 currently not possible to run ``listIndexes`` from within a transaction.
 
 
+loop
+~~~~
+
+The ``loop`` operation executes sub-operations in a loop.
+It supports the following arguments:
+
+- ``operations``: the sub-operations to run on each loop iteration.
+  Each sub-operation must be a valid operation as described in this
+  specification.
+  
+  If, in the course of executing sub-operations, a sub-operation produces
+  an error or a failure, the test runner MUST NOT execute subsequent
+  sub-operations in the same loop iteration. Depending on the presence of
+  ``storeErrorsAsEntity`` and ``storeFailuresAsEntity`` options, as
+  described below, either the iteration MUST complete and the next iteration
+  MUST be started, or the entire ``loop`` operation MUST complete with the
+  produced failure or error.
+
+- ``storeErrorsAsEntity``: if specified, the runner MUST handle errors
+  arising during sub-operation execution and append a document with error
+  information to the array stored in the specified entity. If
+  ``storeFailuresAsEntity`` is specified, the runner MUST NOT include
+  failures in the errors, to the extent the test runner distinguishes
+  errors and failures as further described below.
+  If ``storeFailuresAsEntity`` is not specified,
+  the runner MUST include failures in the errors. The error document
+  MUST contain the following fields:
```

Comment:
If we explicitly discuss loop behavior on error/failure in an earlier paragraph, I think we can simplify the text here.

> if specified, the test runner MUST capture errors arising during sub-operation execution and append a document with its information to an array stored in the specified entity. If the entity does not exist, the test runner MUST initialize it as an empty array before appending the error document. If the entity does exist and is not an array, the test runner MUST raise an error.
>
> If this option is specified and `storeFailuresAsEntity` is not, failures must also be captured and appended to the array.
>
> Documents appended to the array MUST contain the following fields:

I changed "handle" to "capture" since the current behavior is basically suppressing the error/failure from interrupting the greater test. Also added a note about checking the type of the entity before we try to append (akin to safeguards we have in `saveResultAsEntity`). I didn't bother to distinguish the possible error for a non-array type from the errors we catch here, as I think it's clear that's not a sub-operation error that might be caught.